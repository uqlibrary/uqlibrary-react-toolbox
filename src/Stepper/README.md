This is just some responsive CSS for the stepper component.