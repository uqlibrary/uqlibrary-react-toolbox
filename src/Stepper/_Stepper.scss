.Stepper {
  width: 100%;
  max-width: 1130px;
  margin: 24px auto;

  // Stepper: Targets the div wrapping just the stepper
  > div:first-child {
    overflow: hidden;
  }

  // Targets the step span which contains the icon and text
  > div > div > span {
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
  }
}

// Stepper - responsive css to hide the text on mobile, and just leave the steps
@include mobile {
  .Stepper div span span {
    padding: 0 4px 0 0 !important;
    z-index: 10;
    zoom: 1.5;
  }
  .Stepper div div span {
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
    text-indent: -999px;
    padding: 0px 0px 0px 0px;
  }
}