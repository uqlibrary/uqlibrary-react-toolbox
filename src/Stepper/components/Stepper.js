import React from 'react';
import PropTypes from 'prop-types';
import {Step, Stepper, StepLabel} from 'material-ui/Stepper';

function CustomStepper({activeStep, steps}) {
    return (
        <div className="Stepper">
            <Stepper activeStep={activeStep} style={{padding: '0', margin: '-10px auto'}}>
                {
                    steps.map((step, index) => {
                        return (
                            <Step key={`stepper_${index}`}>
                                <StepLabel
                                    style={{textOverflow: 'ellipsis', overflow: 'hidden'}}>
                                    {step.label}
                                </StepLabel>
                            </Step>
                        );
                    })
                }
            </Stepper>
        </div>
    );
}

CustomStepper.propTypes = {
    activeStep: PropTypes.number,
    steps: PropTypes.array
};

export default CustomStepper;
