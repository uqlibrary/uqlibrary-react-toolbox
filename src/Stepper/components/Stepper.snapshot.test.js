import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import Stepper from './Stepper';

function setup(values) {
    return shallow(<Stepper {...values}/>);
}

describe('Add record stepper tests', () => {
    it('should render steps and activate step 2', () => {
        const values = {
            activeStep: 1,
            steps: [
                {label: 'Step 1'},
                {label: 'Step 2'},
                {label: 'Step 3'}
            ]
        };
        const wrapper = setup(values);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
