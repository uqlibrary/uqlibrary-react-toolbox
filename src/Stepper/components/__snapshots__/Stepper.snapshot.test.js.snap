// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add record stepper tests should render steps and activate step 2 1`] = `
<div
  className="Stepper"
>
  <Stepper
    activeStep={1}
    connector={<pure(StepConnector) />}
    linear={true}
    orientation="horizontal"
    style={
      Object {
        "margin": "-10px auto",
        "padding": "0",
      }
    }
  >
    <Step
      key="stepper_0"
    >
      <StepLabel
        style={
          Object {
            "overflow": "hidden",
            "textOverflow": "ellipsis",
          }
        }
      >
        Step 1
      </StepLabel>
    </Step>
    <Step
      key="stepper_1"
    >
      <StepLabel
        style={
          Object {
            "overflow": "hidden",
            "textOverflow": "ellipsis",
          }
        }
      >
        Step 2
      </StepLabel>
    </Step>
    <Step
      key="stepper_2"
    >
      <StepLabel
        style={
          Object {
            "overflow": "hidden",
            "textOverflow": "ellipsis",
          }
        }
      >
        Step 3
      </StepLabel>
    </Step>
  </Stepper>
</div>
`;
